### DISCUSSION TITLE
Invoking PubSub on @AfterInsert()

### DISCUSSION THREAD

**Discussion by ghost** [MichalLytek/type-graphql#1132] [Q&A] [answered]
Hello everyone,

I was wondering if it is possible to emit a pubsub event from an @AfterInsert()

Simple example:

```typescript
@Entity()
@ObjectType()
export class User extends BasicEntity {
  @Field(() => String)
  @Column({ comment: 'User First Name', nullable: false })
  firstName!: string;

  @AfterInsert()
  async publishEvent(@PubSub() pubsub: PubSubEngine) {
    await pubsub.publish('NEW_USER', this);
  }
}
```

This currently does not work and I am wondering if it is possible some other way! 

Thanks,
Connor

**Comment by MichalLytek** [comment, marked as answer]
Use `PubSub` from `graphql-subscriptions` directly: `export const pubSub = new PubSub();`
Remember to register the custom pubsub - search in docs.

**Reply by ghost** [reply]
Thanks for getting back to me so quickly Michal. I literally just realized It was very easy to do.

For anyone wondering, its as simple as:

```typescript

import { pubsub } from '..<path to custom pubsub>'

@Entity()
@ObjectType()
export class User extends BasicEntity {
  @Field(() => String)
  @Column({ comment: 'User First Name', nullable: false })
  firstName!: string;

  @AfterInsert()
  async publishEvent() {
    await pubsub.publish('NEW_USER', this);
  }
}
```

**Reply by sgentile** [reply]
@CMethoddata this is not too helpful - like it doesn't show the custom pubsub, nor shows how you registered it ?

**Reply by ghost** [reply]
@sgentile It is already in the docs mentioned above by @MichalLytek. https://typegraphql.com/docs/subscriptions.html#using-a-custom-pubsub-system

**Reply by sgentile** [reply]
I solved my issue by registered pubsub with typedi and injecting it.   I'm not creating a custom pubsub, already using RedisPubSub just wanted to inject the pubsub in a service vs. the resolver.   Thank you !
